source/collections/Queue.ooc

```diff
@@ -0,0 +1,51 @@
+
+Queue: class <T> {
+	_queue: T*
+	_capacity: Int
+	capacity ::= this _capacity
+	_count: Int
+	count ::= this _count
+	_head: Int
+	_tail: Int
+
+	init: func (=_capacity) {
+		this _queue = gc_calloc(capacity as SizeT, T size)
+		this _head = 0
+		this _tail = 0
+	}
+
+	free: override func {
+		gc_free(_queue)
+		super()
+	}
+
+	isEmpty: func -> Bool {
+		this _count == 0
+	}
+
+	isFull: func -> Bool {
+		this _count == this _capacity
+	}
```

Comment:
`isEmpty` and `isFull` should be properties, like `count` and `capacity` above.

Also, by order from above, we don't put `is` in property names if it can be avoided, so
```ooc
empty ::= this _count == 0
full ::= this _count == this _capacity
```